front/src/applications/operationalStudies/components/MacroEditor/ngeToOsrd.ts

```diff
@@ -683,4 +671,40 @@ const handleOperation = async ({
   }
 };
 
-export default handleOperation;
+export const convertNgeDtoToOsrd = (dto: NetzgrafikDto) => {
+  const macroNodes: MacroNodeForm[] = [];
+  for (const node of dto.nodes) {
+    macroNodes.push({
+      ...castNgeNode(node, dto.labels),
+      path_item_key: `trigram:${node.betriebspunktName}`,
+    });
+  }
+
+  const trainSchedules: TrainSchedule[] = [];
+  const pacedTrains: PacedTrain[] = [];
+  for (const trainrun of dto.trainruns) {
+    const { path, labels, startDate, schedule } = generateTrainrunProperties(dto, trainrun);
```

Comment:
Another point is that the import function is very strict, on the quality of the file imported. Here I try to import [this network graphic](https://schweizerischebundesbahnen.github.io/netzgrafik-editor-frontend/)

The current state of this feature expects a file without the potential errors of `getTrainrunSectionsByTrainrunId()`:
- not continuous trainrun
- missing departure section

which is very restrictive, either accepting or refusing a whole file : would it be better to allow to import as much trains as possible, and skip the errors?

```suggestion
  for (const trainrun of dto.trainruns) {
    if (!getTrainrunSectionsByTrainrunId(dto, trainrun.id)) continue;
    const { path, labels, startDate, schedule } = generateTrainrunProperties(dto, trainrun);
```

And allowing the main func to return null instead (or something similar):
```ts
  // if (orderedSections.length !== sections.length) {
  //   throw new Error('Trainrun is not continuous');
  // }
  if (orderedSections.length !== sections.length) {
    console.error(`Trainrun ${trainrunId} is not continuous`);
    return null;
  }
```

Reply:
I would prefer to be as strict as possible for now, and incrementally work our way towards handling more files by enhancing `ngeToOsrd`.

I do agree that we'll never be able to support e.g. disjoint trainruns, so a mechanism to report to the user which trains failed the import would be nicer. But I think informing the user is important here.